### ISSUE TITLE
.cuda() is so slow that is slower than work in cpu

### ISSUE THREAD

**Issue by McGeeForest** [pytorch/pytorch#59366]
## hello, I'm trying to use torch in CUDA environment.(windows 10 )But it runs too SLOW

**my GPU is 3090*2.**

my CUDA version is `456` and my cuda toolkit version is `11.3`. 

So I install the cudNN which version is `8.2` like below :

![image](https://user-images.githubusercontent.com/35029370/120608993-596d6200-c484-11eb-99a0-c6334855490c.png)

when I run the code:
`testensor = torch.FloatTensor([1.0, 2.0, 3.0]).cuda()`

it is so SLOW.  But  `testensor = torch.FloatTensor([1.0, 2.0, 3.0])` is quickly.

What should I do? Which I configed is wrong?

And my torch version is `1.8.1`. I install it by the below wey.

![image](https://user-images.githubusercontent.com/35029370/120609936-44dd9980-c485-11eb-9073-ab8570adf3dd.png)

cc @malfet @seemethere @walterddr @ngimel @peterjc123 @mszhanyi @skyline75489 @nbcsm

**Comment by skyline75489** [comment]
Hi @McGeeForest . It's expected that if you do `.cuda()` for the first time after launch, it might be slower because of the device initialization & warmup. Can you try `.cuda()` again to see if it helps?

Also, a simple `.cuda()` operation can't really tells you the full picture of GPU performance, you might want to do some real-life training to see how well GPU behaves.

**Comment by McGeeForest** [comment]
> Hi @McGeeForest . It's expected that if you do `.cuda()` for the first time after launch, it might be slower because of the device initialization & warmup. Can you try `.cuda()` again to see if it helps?
> 
> Also, a simple `.cuda()` operation can't really tells you the full picture of GPU performance, you might want to do some real-life training to see how well GPU behaves.

Thank you for your reply. When I was training, I found that the speed of tensor transferring to GPU is too slow. It may be hundreds of times slower than running training only on CPU. Maybe it is the problem of various drivers?

**Comment by peterjc123** [comment]
> **my GPU is 3090*2.**
> 
> my CUDA version is `456` and my cuda toolkit version is `11.3`.

I guess PyTorch built with CUDA 11.2+ would improve the situation. But sadly, the binaries are not available yet.

**Comment by peterjc123** [comment]
BTW, I install PyTorch 1.8.1 built with CUDA 11.1 on a PC with GTX 3080 and I cannot reproduce your problem.
```cmd
(base) C:\Users\peter>conda list pytorch
# packages in environment at C:\Users\peter\anaconda3:
#
# Name                    Version                   Build  Channel
pytorch                   1.8.1           py3.8_cuda11.1_cudnn8_0    pytorch

(base) C:\Users\peter>nvidia-smi
Thu Jun  3 21:05:04 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 466.47       Driver Version: 466.47       CUDA Version: 11.3     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ... WDDM  | 00000000:01:00.0  On |                  N/A |
|  0%   51C    P8    17W / 320W |   1007MiB / 10240MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

```py
>>> import torch
>>> import time
>>> st = time.time(); testensor = torch.FloatTensor([1.0, 2.0, 3.0]).cuda(); print(time.time() - st, 's')
2.002736806869507 s
>>> torch.__version__
'1.8.1'
```

**Comment by peterjc123** [comment]
A potential cause could be the inconsistency of the CUDA libs used when installed by conda and pip. When using pip, we will install cuda 11.1.0 libs while conda tries to pull 11.1.1 libs. 
@McGeeForest Could you please do an experiment? Just copying the CUDA libs from your local installation to the PyTorch installation (typically at `[Python root]\Lib\site-packages\torch\lib`) and see if the problem is resolved.

**Comment by prova** [comment]
On WSL2, latest driver, I get 1.6 s with the first run and 0.005-6 s with CUDA repeating many times over, and 0.0001 s with CPU.

So still, CPU beats GPU by about 5-6 times.

**Comment by kshitij12345** [comment]
```python
testensor = torch.FloatTensor([1.0, 2.0, 3.0]) # Creating a tensor on CPU
testensor = torch.FloatTensor([1.0, 2.0, 3.0]).cuda() # Creating a tensor on CPU and copying it to CUDA
```

Second will be certainly slower than first due to more work being done. However I am not sure about how slow it should be.

**Comment by mszhanyi** [comment]
@prova, I don't it is a good example to compare performance between CPU and GPU because the computation is too small.
GPU is throughput oriented and CPU is latency oriented.
One CPU CORE's main frequency is higher than ONE GPU CORE's, but GPU has much more cores than CPU.
CPU is faster on 1000 * 1000 and GPU's performance is much better on range(1000) * range(1000)

**Comment by prova** [comment]
For sure, jumped on it just having checked. Would be good to have a good set of (fair) performance comparison tests.

**Comment by seemethere** [comment]
> @prova, I don't think it is a good example to compare performance between CPU and GPU because the computation is too small for parallel.
> GPU is throughput oriented and CPU is latency oriented.
> One CPU CORE's main frequency is higher than ONE GPU CORE's, but GPU has much more cores than CPU.
> CPU is faster on 1000 * 1000 and GPU's performance is much better on range(1000) * range(1000)

Closing since I think this issue is now resolved by this comment
